OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -211,5 +220,14 @@ public FrozenActor FromID(uint id)
 
 			return ret;
 		}
+
+		public IEnumerable<Pair<CPos, Color>> RadarSignatureCells(Actor self)
+		{
+			// Filter only actors that have been discovered by the player
+			// HasRenderables is only True once the actor enters the viewport, so is not enough for the minimap
+			foreach (var frozenActor in frozen.Values.Where(f => f.NeedRenderables || f.HasRenderables))
+				foreach (var cell in frozenActor.OccupiedCells)
+					yield return Pair.New(cell, frozenActor.RadarColor);
```

Comment:
How about you call `RadarSignatureCells` on the original actor when it is visible, and cache what that returns instead?